src/components/Ido/Ido.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react'
+import { Grid, Typography, Box, CardMedia } from '@material-ui/core'
+import { SwapToken } from '@components/Swap/Swap'
+import useStyle from './style'
+import AnimatedButton from '@components/AnimatedButton/AnimatedButton'
+import icons from '@static/icons'
+import classNames from 'classnames'
+import { printBN } from '@consts/utils'
+import DepositAmountInput from '@components/Inputs/DepositAmountInput/DepositAmountInput'
+
+interface IdoInterface {
+  xBtc: string
+  xEth: string
+  sol: string
+  usd: string
+  xUsd: string
+  header: string
+  buttonHeader: string
+  tokens: SwapToken[]
+}
+
+const Ido: React.FC<IdoInterface> = ({
+  header,
+  xUsd,
+  usd,
+  sol,
+  xEth,
+  xBtc,
+  buttonHeader,
+  tokens
+}) => {
+  const [amountFrom, setAmountFrom] = React.useState<string>('')
+  const [tokenIndex] = React.useState<number | null>(tokens.length ? 0 : null)
```

Comment:
get rid of this, just set balanve and decimals through props

Reply:
done